### DISCUSSION TITLE
PHP Fatal Error during Install Cache Clearing

### DISCUSSION THREAD

**Discussion by twistedsymphony** [kimai/kimai#4806] [On-Premise: Installation & Updates] [answered]
I'm attempting to install Kimai on a locally hosted server so my business can try it out. It's a freshly created server using a bitnami LAMP image. This is running PHP 8.2 and has all of the required packages enabled.

I followed the SSH instructions here: https://www.kimai.org/documentation/installation.html
and everything went well until I got to the step with "bin/console kimia:install -n"
this command appears to run fine until it gets to the rebuilding of the cache where it throws this error:

```
Rebuilding your cache, please be patient ...

 // Clearing the cache for the prod environment with debug false

PHP Fatal error:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 20480 bytes) in /opt/bitnami/apache/htdocs/kimai/vendor/twig/twig/src/Node/Node.php on line 158
PHP Fatal error:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 24576 bytes) in /opt/bitnami/apache/htdocs/kimai/vendor/composer/ClassLoader.php on line 576
zend_mm_heap corrupted
Aborted
```

the remaining steps work fine but attempting to visit the site it throws an Error 500. checking the /var/log/prod_log only shows the above cache errors (nothing related to the error 500)

I've redoubled the file permission steps to no avail.
I've tried running "bin/console kimai:reload --env=prod" which throws the same cache errors. 
and even started from scratch with a fresh server install and still the results are the same.

I'm out of ideas, any advice is appreciated.

**Comment by niels-przybilla** [comment, marked as answer]
Good day twistedsymphony,
it looks like you need to give your PHP more memory in php.ini
Try at least 256M if not 512M
Best wishes

Niels

**Reply by twistedsymphony** [reply]
Increasing to 512M and the cache errors have stopped, however the site still throws an error 500 with no new errors appearing in the logs.
I've tested that php and apache error logging work by creating a php file with errors in and and can confirm that the errors show up in the log. Similarly php code seems to be executing appropriately when there are no errors in the files I've manually created.

**Comment by niels-przybilla** [comment]
Did you check permissions? 

Adjust file permission
You have to allow PHP (your webserver process) to write to var/ and it subdirectories.

Here is an example for Debian/Ubuntu (to be executed inside the Kimai directory):

chown -R :www-data .
chmod -R g+r .
chmod -R g+rw var/

**Reply by twistedsymphony** [reply]
Yes these are in the instructions I followed and I've confirmed that ownership and permissions were changed appropriately to all of them.

As a test I just tried setting to chmod 777 on the var folder and that worked... doing a little digging it seems that bitnami uses the user "daemon" to execute php-fpm code so rolling back to the suggested chmod of g+rw on var/ I got it to work by simply adding daemon to the www-data group with:

sudo usermod -a -G www-data daemon

now everything appears to work. 

thanks for all of your help!
